sway-core/src/type_system/type_info.rs

```diff
@@ -528,7 +546,7 @@ impl TypeInfo {
         match self {
             Unknown => "unknown".into(),
             UnknownGeneric { name, .. } => name.to_string(),
-            Str(x) => format!("str[{}]", x),
+            Str(x) => format!("str[{}]", x.val),
```

Comment:
Now that we have `.val()`, I think we should make the fields on `Length` private and use the method everywhere. Same for `span()`.

Reply:
Oh I've done that already, but `Length` lives in `type_info.rs` so `x.val` still works here, so I missed this one. I guess the right thing to do would be to move `Length` into a separate module.